d3-dag.d.ts

```diff
@@ -0,0 +1,781 @@
+                ) : this
+                /** returns the current layering accessor, which defaults to d3.layeringSimplex(). */
+                layering():
+                    Layering.Simplex|
+                    Layering.CoffmanGraham|
+                    Layering.LongestPath|
+                    Layering.Topological
+                /** sets the decross accessor to the specified function.
+                 * A decross accessor takes a dag as an array of layers where each layer is an array of nodes,
+                 * and modifies the order of nodes in each layer to reduce the number of link crossings.*/
+                decross(decross: 
+                    DeCross.Optimal|
+                    DeCross.TwoLayer
+                ) : this
+                /** returns the current decross accessor, which defaults to d3.decrossTwoLayer(). */
+                decross():
+                    DeCross.Optimal|
+                    DeCross.TwoLayer
+                /** sets the coord accessor to the specified function.
+                * A coord accessor takes a dag as an array of layers where each layer is an array of nodes
+                * and a separation function, which takes adjacent nodes and specifies their relative separation.
+                * The coord accessor assigns every node an x property in [0, 1] to specify the actual layout. */
+                coord(coord: 
+                    Coord.Center|
+                    Coord.Vert|
+                    Coord.MinCurve|
+                    Coord.Greedy|
+                    Coord.Topological
+                ): this
+                /** returns the current coord accessor, which defaults to d3.coordGreedy(). */
+                coord(): 
+                    Coord.Center|
+                    Coord.Vert|
+                    Coord.MinCurve|
+                    Coord.Greedy|
+                    Coord.Topological
+
+                /** sets the separation accessor to the specified function.
+                 * The separation accessor function takes two adjacent dag nodes and sets their relative separation,
+                * thus any constant function will produce the same results.
+                * Another other common setting is: `(a, b) => (a.data !== undefined) + (b.data !== undefined)`
+                * which will wrap edges around nodes, but give them no spaceing themselves.
+                */
+                separation(separation: Separator<NodeDatum, LinkDatum>): this
+                /** returns the current separation accessor, which defaults to: `(a, b) => 1` */
+                separation(): Separator<NodeDatum, LinkDatum>
+            }
+
+            /** Sugiyama Provided Layering Algorithm */
+            export namespace Layering {
+
+                // A layering accessor takes a dag and assigns every node a layer attribute
+                // from zero to the number of layers
+                interface Accessor {():{_layeringSignature_}} // opaque typing

```

Comment:
Your exposing my newness to typescript, but I don't understand what this is saying, and `opaque typing` doesn't fill in the cracks in my knowledge.